examples/onoff_light_bt/src/comm.rs

```diff
@@ -0,0 +1,259 @@
+/*
+ *
+ *    Copyright (c) 2020-2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+use embassy_sync::blocking_mutex::raw::NoopRawMutex;
+use log::{error, info, warn};
+
+use rs_matter::data_model::objects::{
+    AttrDataEncoder, AttrDataWriter, AttrDetails, AttrType, CmdDataEncoder, CmdDetails, Dataver,
+    Handler, NonBlockingHandler,
+};
+use rs_matter::data_model::sdm::nw_commissioning::{
+    AddWifiNetworkRequest, Attributes, Commands, ConnectNetworkRequest, ConnectNetworkResponse,
+    NetworkCommissioningStatus, NetworkConfigResponse, RemoveNetworkRequest, ReorderNetworkRequest,
+    ResponseCommands, ScanNetworksRequest, WIFI_CLUSTER,
+};
+use rs_matter::error::{Error, ErrorCode};
+use rs_matter::interaction_model::core::IMStatusCode;
+use rs_matter::interaction_model::messages::ib::Status;
+use rs_matter::tlv::{FromTLV, OctetStr, TLVElement};
+use rs_matter::transport::exchange::Exchange;
+use rs_matter::utils::notification::Notification;
+use rs_matter::utils::rand::Rand;
+
+/// A _fake_ cluster implementing the Matter Network Commissioning Cluster
+/// for managing WiFi networks.
+///
+/// We only pretend to manage these for the purposes of the BT demo.
+pub struct WifiNwCommCluster<'a> {
+    data_ver: Dataver,
+    nw_setup_complete: &'a Notification<NoopRawMutex>,
+}
+
+impl<'a> WifiNwCommCluster<'a> {
+    /// Create a new instance.
+    pub fn new(rand: Rand, nw_setup_complete: &'a Notification<NoopRawMutex>) -> Self {
+        Self {
+            data_ver: Dataver::new(rand),
+            nw_setup_complete,
+        }
+    }
+
+    /// Read an attribute.
+    pub fn read(
+        &self,
+        attr: &AttrDetails<'_>,
+        encoder: AttrDataEncoder<'_, '_, '_>,
+    ) -> Result<(), Error> {
+        if let Some(mut writer) = encoder.with_dataver(self.data_ver.get())? {
```

Comment:
> I think at some point we need more documentation. I was trying to follow return types for with_dataver to figure out why it may return None or not. As I understand it we return None if the data version passed in is * W.r.t. docu: the APIs you are looking at are pre-2024. Yes, they need documentation. We now have [this](https://github.com/project-chip/rs-matter/issues/191)
* W.r.t. early returns being easier - OK I'll make early returns

Reply:
Done!